docs/sdk/src/reference_docs/runtime_vs_smart_contract.rs

```diff
@@ -1,6 +1,205 @@
+//! *DApp*, then write a Smart Contract.
 //!
-//! Why one can be weighed, and one MUST be metered.
-//! https://forum.polkadot.network/t/where-contracts-fail-and-runtimes-chains-are-needed/4464/3
+//! This is a comparative analysis of Substrate-based Runtimes and Smart Contracts, highlighting
+//! their main differences. Our aim is to equip you with a clear understanding of how these two
+//! methods of deploying on-chain logic diverge in their design, usage, and implications.
+//!
+//! Both Runtimes and Smart Contracts serve distinct purposes. Runtimes offer deep customization for
+//! blockchain development, while Smart Contracts provide a more accessible approach for
+//! decentralized applications. Understanding their differences is crucial in choosing the right
+//! approach for a specific solution.
+//!
+//! ## Substrate
+//! Substrate is a modular framework that enables the creation of purpose-specific blockchains. In
+//! the Polkadot ecosystem you can find two distinct approaches for on-chain code execution:
+//! [Runtime Development](#runtime-in-substrate) and [Smart Contracts](#smart-contracts).
+//!
+//! #### Smart Contracts in Substrate
+//! Smart Contracts are autonomous, programmable constructs deployed on the blockchain.
+//! In [FRAME](frame), Smart Contracts infrastructure is implemented by the
+//! [`pallet_contracts`](../../../pallet_contracts/index.html) for WASM-based contracts or the
+//! [`pallet_evm`](../../../pallet_evm/index.html) for EVM-compatible contracts. These pallets
+//! enable Smart Contract developers to build applications and systems on top of a Substrate-based
+//! blockchain.
+//!
+//! #### Runtime in Substrate
+//! The Runtime is the state transition function of a Substrate-based blockchain. It defines the
+//! rules for processing transactions and blocks, essentially governing the behavior and
+//! capabilities of a blockchain.
+//!
+//! ## Comparative Table
+//!
+//! | Aspect                | Runtime                                                                 | Smart Contracts                                                      |
+//! |-----------------------|-------------------------------------------------------------------------|----------------------------------------------------------------------|
+//! | **Design Philosophy** | Core logic of a blockchain, allowing broad and deep customization.      | Designed for DApps deployed on the blockchain runtime.|
+//! | **Development Complexity** | Requires in-depth knowledge of Rust and Substrate. Suitable for complex blockchain architectures.         | Easier to develop with knowledge of Smart Contract languages like Solidity or [ink!](https://use.ink/). |
+//! | **Upgradeability and Flexibility** | Offers seamless upgradeability, allowing entire blockchain logic modifications without hard forks. | Less flexible in upgrading but offers more straightforward deployment and iteration. |
```

Comment:
Not sure this is true, you can also invoke governance to update the code of a contract through the set_code root dispatcheable, or define a contract that let some kind of admin account invoke the set_code_hash function maybe we can link to https://use.ink/5.x/basics/upgradeable-contracts

I think the flexibility rather comes from the fact that you can define migration logic when you write a runtime, this is what makes in my view the upgreadibility more flexible